.github/workflows/ci.yml

```diff
@@ -22,11 +22,20 @@ jobs:
         with:
           node-version: 16.14.2
 
-      - name: Install dependencies
+      - name: Install only prod dependencies
+        run: npm install --production
+
+      - name: Try compiling typescript with prod dependencies (matches EB behavior)
+        run: npm run build
+
+      - name: Install all dependencies (for CI)
         run: npm install
 
       - name: Compile Typescript
         run: npm run build
 
       - name: Check code formatting
         run: npx prettier --check .
+
+      - name: Run tests
```

Comment:
I wouldn't include this for now. We'd need to add postgres and all it's setup first.